prefect_dask/task_runners.py

```diff
@@ -294,8 +295,18 @@ async def _start(self, exit_stack: AsyncExitStack):
             self._connect_to = self._cluster = await exit_stack.enter_async_context(
                 self.cluster_class(asynchronous=True, **self.cluster_kwargs)
             )
+
+            # If used with the operator implementation of KubeCluster,
+            # the cluster is not automatically started
+            if self._cluster.status.value != "running":
+                await self._cluster._start()
```

Comment:
You're right, that does not seem like the intended way of starting a cluster but I didn't find something relevant in [their documentation](https://kubernetes.dask.org/en/latest/operator_kubecluster.html) when used with the `asynchronous` arg. If we create the instance without it, this `_start` method is called. I asked in their forum [here](https://dask.discourse.group/t/new-operator-kubecluster-and-speccluster/1212) to better understand how to use the new implementation asynchronously.

Reply:
[In their `__init__`:](https://kubernetes.dask.org/en/latest/_modules/dask_kubernetes/operator/kubecluster/kubecluster.html#KubeCluster)
```
        if not self.asynchronous:
            self._loop_runner.start()
            self.sync(self._start)
```

Reply:
Yes but prefect forces the asynchronous arg to `True`. Should we allow to pass `False` when used with `operator.KubeCluster`?

Reply:
It's weird that they don't call `_start()` in the `__aenter__` method when we enter the context above. This seems like an oversight. It feels like we should report this upstream and we should get some clarity on their intent here instead of introducing workarounds.

Reply:
Sure! Let's wait to see if they answer to the thread I created in their discourse.

Reply:
If you don't hear anything back, it might be faster to submit a GitHub issue.

Reply:
You're right. I'll try to figure out how they should handle the asynchronous start.